streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -255,6 +259,14 @@ public Subscription subscription(Set<String> topics) {
             state.capacity = state.capacity + 1d;
         }
 
+        // Source nodes are build ahead of time, but it's at this
+        // point we see the topics subscribed to
+        if (streamThread.builder.sourceTopicPattern() != null) {
+            SubscriptionUpdates subscriptionUpdates = streamThread.builder.getSubscriptionUpdates();
+            subscriptionUpdates.addTopicUpdates(regexSubscribedTopics);
```

Comment:
Should we "replace" the topic list instead of just adding to the list? Also, how about changing the function name to "updateTopics" since we should not "add" but replace here, right?